templates/types/streaming/nextjs/app/api/embed/embeddings.ts

```diff
@@ -0,0 +1,28 @@
+import { Document, MetadataMode, Settings, SimpleNodeParser } from "llamaindex";
+import pdf from "pdf-parse";
+
+export async function splitAndEmbed(document: string) {
+  const nodeParser = new SimpleNodeParser({
+    chunkSize: Settings.chunkSize,
+    chunkOverlap: Settings.chunkOverlap,
+  });
+  const nodes = nodeParser.getNodesFromDocuments([
+    new Document({ text: document }),
+  ]);
+  const texts = nodes.map((node) => node.getContent(MetadataMode.EMBED));
+  const embeddings = await Settings.embedModel.getTextEmbeddingsBatch(texts);
+  return nodes.map((node, i) => ({
+    text: node.getContent(MetadataMode.NONE),
+    embedding: embeddings[i],
+  }));
+}
+
+export async function getPdfDetail(rawPdf: string) {
+  const pdfBuffer = Buffer.from(rawPdf.split(",")[1], "base64");
+  const content = (await pdf(pdfBuffer)).text;
+  const embeddings = await splitAndEmbed(content);
+  return {
+    content,
+    embeddings,
+  };
```

Comment:
The readers will return a `Document` you can then use:

```
const pipeline = new IngestionPipeline({
    transformations: [
      new SimpleNodeParser({ chunkSize: 1024, chunkOverlap: 20 }),
      new OpenAIEmbedding(),
    ],
  });

  // run the pipeline
  const nodes = await pipeline.run({ documents: [document] });
 ```

in the `splitAndEmbed` function.

This will return `TextNode[]` objects with contain text content and embeddings (what you actually want)
This means you can directly return

```suggestion
  const nodes = await splitAndEmbed(content);
  return nodes;
```